static/js/containers/DashboardPage.js

```diff
@@ -78,23 +82,32 @@ class DashboardPage extends React.Component {
 
     let courseKey = query.course_key;
     if (query.status === 'receipt') {
-      let course = {};
+      let course = {}, courseRun = {};
       for (let program of dashboard.programs) {
         for (let courseInProgram of program.courses) {
-          for (let courseRun of courseInProgram.runs) {
-            if (courseRun.course_id === courseKey) {
+          for (let courseRunInProgram of courseInProgram.runs) {
+            if (courseRunInProgram.course_id === courseKey) {
               course = courseInProgram;
+              courseRun = courseRunInProgram;
             }
           }
         }
       }
 
-      dispatch(setToastMessage({
-        title: 'Order Complete!',
-        message: `You are now enrolled in ${course.title}`,
-        icon: TOAST_SUCCESS,
-      }));
-      this.context.router.push('/dashboard/');
+      if (courseRun.status === STATUS_CAN_UPGRADE || courseRun.status === STATUS_OFFERED) {
+        dispatch(setToastMessage({
+          title: 'Order is processing',
+          message: `Please refresh after a minute...`,
+        }));
+        dispatch(updateCourseStatus(courseRun.course_id, STATUS_CURRENTLY_ENROLLED));
```

Comment:
if the payment failed will this be un-done when the user next refreshes the dashboard?